### PR TITLE
build and publish multiple images

### PR DESCRIPTION
**Pull request by thirdeyenick**
This should build and publish the tailscale client, k8s-operator and k8s-nameserver images

### CODE REVIEW

File: .github/workflows/build-and-publish-images.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and publish k8s-operator image
+        env:
+          REPO: ghcr.io/${{ github.repository_owner }}/tailscale/k8s-operator
```
Comment (ctrox) [inline .github/workflows/build-and-publish-images.yml:27]: Hmm does that really work? GitHub registry images are in the format of `ghcr.io/NAMESPACE/IMAGE_NAME` so the image name would need to allow a `/` in it?

Comment (thirdeyenick) [reply .github/workflows/build-and-publish-images.yml:27]: ah, well then it might not work :-) Will change it.

Comment (thirdeyenick) [reply .github/workflows/build-and-publish-images.yml]: changed it. Will something like `tailscale-k8s-operator` work as the image name?

Comment (ctrox) [reply .github/workflows/build-and-publish-images.yml]: I guess we'll just have to test it to find out :smile:

**Review by ctrox — APPROVED** [review]

**Pull request merged**